### PR TITLE
[CLI-2900] Print detailed topic info in `kafka topic list`

### PR DESCRIPTION
**Pull request by MuweiHe**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- Display replication factor and partition count in `confluent kafka topic list`

Bug Fixes
- PLACEHOLDER

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----

References
----------

https://confluentinc.atlassian.net/browse/FF-8833

Test & Review
-------------

Manually tests for both cloud and on-prem.
Added integration tests.

### CODE REVIEW

File: internal/kafka/command_topic_list.go [hunk 1]
```diff
@@ -9,10 +9,18 @@ import (
...
 	Name string `human:"Name" serialized:"name"`
 }
 
+type topicOut struct {
+	Name              string `human:"Name" serialized:"name"`
+	Kind              string `human:"Kind" serialized:"kind"`
+	IsInternal        bool   `human:"Is Internal" serialized:"is_internal"`
+	ReplicationFactor int32  `human:"Replication Factor" serialized:"replication_factor"`
+	PartitionsCount   int32  `human:"Partitions Count" serialized:"partitions_count"`
+}
```
Comment (brianstrauch) [inline internal/kafka/command_topic_list.go:22]: Why not print these new fields by default?

File: internal/kafka/command_topic_list.go [hunk 2]
```diff
@@ -9,10 +9,18 @@ import (
...
 
-type topicOut struct {
+type topicNameOut struct {
 	Name string `human:"Name" serialized:"name"`
 }
 
+type topicOut struct {
+	Name              string `human:"Name" serialized:"name"`
+	Kind              string `human:"Kind" serialized:"kind"`
+	IsInternal        bool   `human:"Is Internal" serialized:"is_internal"`
```
Comment (brianstrauch) [inline internal/kafka/command_topic_list.go:19]: We use the same format for `IsCurrent`
```suggestion
	IsInternal        bool   `human:"Internal" serialized:"is_internal"`
```

File: internal/kafka/command_topic_list.go [hunk 3]
```diff
@@ -9,10 +9,18 @@ import (
...
+	ReplicationFactor int32  `human:"Replication Factor" serialized:"replication_factor"`
+	PartitionsCount   int32  `human:"Partitions Count" serialized:"partitions_count"`
```
Comment (brianstrauch) [inline internal/kafka/command_topic_list.go:21]:
```suggestion
	PartitionCount   int32  `human:"Partition Count" serialized:"partition_count"`
```

File: internal/kafka/command_topic_list.go (hunk 1 above)
Comment (MuweiHe) [reply internal/kafka/command_topic_list.go:22]: yes lets do that

Comment (brianstrauch) [reply internal/kafka/command_topic_list.go]: (Awesome! Also note that `kafka topic describe` is already taken, and we can't change it until: https://confluentinc.atlassian.net/browse/CLI-2695?focusedCommentId=2026491)

File: test/test-server/kafka_rest_router.go [hunk 4]
```diff
@@ -173,6 +173,7 @@ func handleKafkaRestTopics(t *testing.T) http.HandlerFunc {
 					"topic_name": "topic1",
 					"is_internal": false,
 					"replication_factor": 3,
+					"partitions_count": 6,
```
Comment (brianstrauch) [inline test/test-server/kafka_rest_router.go:176]: Can you rewrite this so it isn't a big JSON string?

File: internal/kafka/command_topic_list.go (hunk 3 above)
Comment (MuweiHe) [reply internal/kafka/command_topic_list.go:17]: This aligns with the api response.. it's `topic.GetPartitionsCount()`

Comment (brianstrauch) [reply internal/kafka/command_topic_list.go:17]: Yeah... that's not proper grammar though :(

Comment (brianstrauch) [reply internal/kafka/command_topic_list.go:17]: (Maybe we should open an API ticket to fix this in the next major version of the API, too)

File: test/test-server/kafka_rest_router.go (hunk 4 above)
Comment (MuweiHe) [reply test/test-server/kafka_rest_router.go:176]: How do i rewrite this? This string is being written into the http response: `io.WriteString(w, response)`

File: test/test-server/kafka_rest_router.go [hunk 5]
```diff
@@ -162,35 +162,24 @@ func handleKafkaRestTopics(t *testing.T) http.HandlerFunc {
...
-					"is_internal": true,
-					"replication_factor": 4,
-					"partitions": {"related": "http://localhost:9391/v3/clusters/cluster-1/topics/topic-2/partitions"},
-					"configs": {"related": "http://localhost:9391/v3/clusters/cluster-1/topics/topic-2/configs"},
-					"partition_reassignments": {"related": "http://localhost:9391/v3/clusters/cluster-1/topics/topic-2/partitions/-/reassignments"}
-					}
-				]
-			}`
-			_, err := io.WriteString(w, response)
+			topic1 := cckafkarestv3.TopicData{
```
Comment (brianstrauch) [inline test/test-server/kafka_rest_router.go:165]: No need to individually define these variables, we could combine everything in this branch into one initialization.

File: internal/kafka/command_topic_list.go (hunk 3 above)
Comment (brianstrauch) [reply internal/kafka/command_topic_list.go:17]: ^ Don't forget!

**Review by brianstrauch — APPROVED** [review]

File: internal/kafka/command_topic_list.go [hunk 6]
```diff
@@ -10,7 +10,11 @@ import (
 )
 
 type topicOut struct {
-	Name string `human:"Name" serialized:"name"`
+	Name              string `human:"Name" serialized:"name"`
+	Kind              string `human:"Kind" serialized:"kind"`
```
Comment (brianstrauch) [inline internal/kafka/command_topic_list.go:14]: This should always be "KafkaTopic", so let's leave it out.
```suggestion
```

**Review by brianstrauch — CHANGES_REQUESTED** [review]

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**